test/units/module_utils/facts/tests_date_time.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import pytest
+import datetime
+
+from ansible.module_utils.facts.system.date_time import DateTimeFactCollector
+
+TIMESTAMP = datetime.datetime(2020, 7, 11, 12, 34, 56, 124356)
+
+
+@pytest.fixture
+def fake_now(monkeypatch):
+
+    class FakeNow:
+        @classmethod
+        def now(cls):
+            return TIMESTAMP
+
+    monkeypatch.setattr(datetime, 'datetime', FakeNow)
+
+
+@pytest.fixture
+def date_collector(monkeypatch, fake_now):
+    monkeypatch.setenv('TZ', 'Australia/Melbourne')
+
+    collector = DateTimeFactCollector()
+    data = collector.collect()
+
+    return data
+
+
+def test_date_time_year(date_collector):
+    assert date_collector['date_time']['year'] == '2020'
```

Comment:
Since you only check keys of the same object in the same fashion, you could collapse this using parameterize:
```suggestion
@pytest.mark.parametrize(
    ('fact_name', 'fact_value'),
    (
        ('year', 2020),
        ('weekday', 'Saturday'),
    ),
)
def test_date_time_facts(date_collector, fact_name, fact_value):
    assert date_collector['date_time'][fact_name] == fact_value
```
This will generate a separate test per each param while allowing you to keep the test function logic the same.

https://docs.pytest.org/en/stable/example/parametrize.html

Reply:
Thank you for the tip. I've improved unit tests with parameterize.